src/librustdoc/test.rs

```diff
@@ -205,6 +207,7 @@ fn run_test(
     mut error_codes: Vec<String>,
     opts: &TestOptions,
     edition: Edition,
+    visited_tests: &Mutex<HashMap<String, usize>>,
```

Comment:
Should this be changed to `visited_tests: &Mutex<HashMap<u64, usize>>`?

It is not required to keep the entire `folder_name` in memory a hash would suffice.